### PR TITLE
[Maps] Add action buttons to PreviewSidebar + other fixes

### PR DESCRIPTION
**Pull request by jshoe**
### Description
- Hook up the MapPreviewSidebar sample selections to the action buttons so you can create a collection, view a heatmap, etc.
- Rename mapSelectedSamples and similar to mapPreviewedSamples to clarify which samples are previewed (displayed on the right sidebar) and selected (checkmark checked on the right sidebar).
- Fix a bug where MapPreviewSidebar would stay even if you switch over to the Projects tab.
- Fix a bug where sample notes weren't showing in the DD tables (should be `db_sample.sample_notes` now).
- Update MapTiler style ID since we switched to a new account yesterday.
- Allow MapPreviewSidebar checkbox selections to persist. This was agreed upon behavior (instead of trying to combine selections from the main table and the preview table).
- Fix loading animations on the main page (not on the sidebar yet!).
- Tweak sampleStatus style (ad-hoc request from Jenn/Maria next to me).

### Demo
Jun-05-2019 11-57-57
Screen Shot 2019-06-05 at 11 48 21 AM

### Test and Reproduce
- Pull the branch, go to data discovery samples, go to the map, click a tooltip header to open up the PreviewSidebar, go through the interactions described above.

### CODE REVIEW

File: app/assets/src/components/views/discovery/DiscoveryView.jsx [hunk 1]
```diff
@@ -697,14 +698,14 @@ class DiscoveryView extends React.Component {
   };
 
   handleMapTooltipTitleClick = locationId => {
-    this.setState({ mapSelectedLocationId: locationId }, () =>
-      this.refreshMapSelectedSamples()
+    this.setState({ mapPreviewedLocationId: locationId }, () =>
```
Comment (tfrcarvalho) [inline app/assets/src/components/views/discovery/DiscoveryView.jsx]: Curious about the `selected` -> `preview` name change?

Comment (jshoe) [reply app/assets/src/components/views/discovery/DiscoveryView.jsx]: I wrote this in the description but it's to clarify which samples are previewed (displayed on the right sidebar) and selected (checkmark checked on the right sidebar)

File: app/assets/src/components/views/discovery/discovery_api.js [hunk 2]
```diff
@@ -101,7 +101,7 @@ const processRawSample = sample => {
         sample.details
       )
     },
-    notes: sample.sample_notes,
+    notes: get("db_sample.sample_notes", sample.details),
```
Comment (tfrcarvalho) [inline app/assets/src/components/views/discovery/discovery_api.js]: thanks for fixing this!

File: app/assets/src/components/views/samples/SamplesView.jsx [hunk 3]
```diff
@@ -221,9 +235,18 @@ class SamplesView extends React.Component {
...
 
   renderCollectionTrigger = () => {
-    const { samples } = this.props;
-    const { selectedSampleIds } = this.state;
+    const {
+      currentDisplay,
+      mapSidebarSelectedSampleIds,
+      mapPreviewedSamples
+    } = this.props;
+    let { samples } = this.props;
```
Comment (tfrcarvalho) [inline app/assets/src/components/views/samples/SamplesView.jsx]: Why let?

File: app/assets/src/components/views/samples/SamplesView.jsx [hunk 4]
```diff
@@ -221,9 +235,18 @@ class SamplesView extends React.Component {
...
+    let { selectedSampleIds } = this.state;
```
Comment (tfrcarvalho) [inline app/assets/src/components/views/samples/SamplesView.jsx]: Why let?

File: app/assets/src/components/views/samples/SamplesView.jsx [hunk 5]
```diff
@@ -221,9 +235,18 @@ class SamplesView extends React.Component {
...
 
+    if (currentDisplay === "map") {
+      samples = mapPreviewedSamples;
```
Comment (tfrcarvalho) [inline app/assets/src/components/views/samples/SamplesView.jsx]: I do not understand why samples is changed here... you should not change props.

**Comment by jshoe** [comment]
@tfrcarvalho updated to get rid of the `let`s 👍

**Pull request merged**